tests/rem_auth_tests.cpp

```diff
@@ -1781,4 +1814,84 @@ BOOST_FIXTURE_TEST_CASE( keys_cleanup_test, rem_auth_tester ) {
+
+      auto payload = join( {
+                        string(account_data.begin(), account_data.end()), string(packed_action.begin(), packed_action.end()),
+                        string(action_timestamp_data.begin(), action_timestamp_data.end()), pub_key_to_bin_string(key_pub), payer.to_string()
+                     } );
+      sha256 digest_execaction = sha256::hash(payload);
+      auto signed_by_key_execaction = key_priv.sign(digest_execaction);
+
+      auto account_balance_before = get_balance(account);
+      auto to_account_balance_before = get_balance(to_account);
+
+      execaction(account, act, action_timestamp, key_pub, signed_by_key_execaction, payer, auths_level);
+
+      auto account_balance_after = get_balance(account);
+      auto to_account_balance_after = get_balance(to_account);
+
+      BOOST_REQUIRE_EQUAL(account_balance_before - transfer_quantity, account_balance_after);
+      BOOST_REQUIRE_EQUAL(to_account_balance_before + transfer_quantity, to_account_balance_after);
+
+      // missing authority of proda
+      BOOST_REQUIRE_THROW(execaction(N(prodc), act, action_timestamp, key_pub, signed_by_key_execaction, payer, auths_level),
+                          eosio_assert_message_exception);
+      // missing authority of producer1
+      BOOST_REQUIRE_THROW(execaction(account, act, action_timestamp, key_pub, signed_by_key_execaction, N(producer1), auths_level),
+                          missing_auth_exception);
+
+      auths_level.push_back(permission_level{N(prodc), config::active_name});
+      action act_with_invalid_auth = get_action(N(rem.token), N(transfer), auths_level, abi_ser_token, mvo()
+                                                ("from", account)
+                                                ("to", to_account)
+                                                ("quantity", core_from_string("0.1000"))
+                                                ("memo", "test transfer"));
+
+      // authorization should contain one permission
+      BOOST_REQUIRE_THROW(execaction(account, act_with_invalid_auth, action_timestamp, key_pub, signed_by_key_execaction, payer, auths_level),
+                          eosio_assert_message_exception);
+      // sign payload by another private key
+      crypto::private_key new_key_priv = crypto::private_key::generate();
+      crypto::public_key new_key_pub   = key_priv.get_public_key();
+      auto signed_by_new_key_execaction = new_key_priv.sign(digest_execaction);
+      // Error expected key different than recovered key
+      BOOST_REQUIRE_THROW(execaction(account, act, action_timestamp, key_pub, signed_by_new_key_execaction, payer, auths_level),
+                          crypto_api_exception);
+      // account has no linked application keys
+      BOOST_REQUIRE_THROW(execaction(account, act, action_timestamp, new_key_pub, signed_by_new_key_execaction, payer, auths_level),
+                        crypto_api_exception);
+   } FC_LOG_AND_RETHROW()
```

Comment:
add a test case where **proda** correctly calls **execaction** but the nested action transfers tokens from **prodb** to **prodc**

Reply:
@roman-tik, done.